doc/source/reference/arrays.rst

```diff
@@ -22,8 +22,6 @@ can be found at :ref:`basics.dtypes`.
 =================== ========================== ============================= =============================
 Kind of Data        pandas Data Type           Scalar                        Array
 =================== ========================== ============================= =============================
-TZ-aware datetime   :class:`DatetimeTZDtype`   :class:`Timestamp`            :ref:`api.arrays.datetime`
-Timedeltas          (none)                     :class:`Timedelta`            :ref:`api.arrays.timedelta`
```

Comment:
why would this change?

Reply:
@jbrockmendel Maybe I didn't completely understand #53058. Are we not removing all dt.td64 dtypes? 

I removed the above, as it was part of the dt/td64 documentation

Reply:
As I said here https://github.com/pandas-dev/pandas/pull/53439#issuecomment-1570711319, we aren't deprecating all dt64/td64, just the un-supported resolutions

Reply:
@jbrockmendel Ah okay understood! 

Looking at the implementation, it looks like the changes for this PR need to be made either in `_ensure_nanosecond_dtype()` in cast.py, or in the implementation for `is_supported_unit()` in dtypes.pyi. 

Seeing that `is_supported_unit() in dtypes.pyi is an interface, I'm assuming that there is a definition for this function for pd.array, but I don't see it. Would you be able to shed some light on how this interface method is being used?